text/0000-rustdoc-json.md

```diff
@@ -0,0 +1,1124 @@
+`type`        | [Type](#Type) | The type of this constant.
+`expr`        | String   | The stringified expression of this constant.
+`value`       | String   | (*Optional*) The value of the evaluated expression for this constant, which is only computed for numeric types.
+`is_literal`  | bool     | Whether this constant is a bool, numeric, string, or char literal.
+
+### `kind == "static"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this static.
+`expr`        | String   | The stringified expression that this static is assigned to.
+`mutable`     | bool     | Whether this static is mutable.
+
+### `kind == "typedef"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type on the right hand side of this definition.
+`generics`    | [Generics](#Generics) | Any generic parameters on the left hand side of this definition.
+
+### `kind == "opaque_ty"`
+
+Represents [trait aliases](https://doc.rust-lang.org/beta/unstable-book/language-features/trait-alias.html)
+of the form:
+
+```rust
+type Foo<T> = Clone + std::fmt::Debug + Into<T>;
+```
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`bounds`      | [[GenericBound](#GenericBound)] | The trait bounds on the right hand side.
+`generics`    | [Generics](#Generics) | Any generic parameters on the type itself.
+
+### `kind == "foreign_type"`
+
+`inner` contains no fields. This item represents a type declaration in an extern block (see [here](https://github.com/rust-lang/rfcs/blob/master/text/1861-extern-types.md)
+for more details):
+
+```rust
+extern {
+    type Foo;
+}
+```
+
+### `kind == "extern_crate"`
+
+Name     | Type     | Description
+---------|----------|-------------------------------------------------------------------------
+`name`   | String   | The name of the extern crate.
+`rename` | String   | (*Optional*) The renaming of this crate with `extern crate foo as bar`.
+
+### `kind == "import"`
+
+Name     | Type     | Description
+---------|----------|-------------------------------------------------------------------------
+`source` | String   | The full path being imported (e.g. `"super::some_mod::other_mod::Struct"`).
+`name`   | String   | The name of the imported item (may be different from the last segment of `source` due to import renaming: `use source as name`).
+`id`     | [ID](#ID) | (*Optional*) The ID of the item being imported.
+`glob`   | bool     | Whether this import ends in a glob: `use source::*`.
+
+### `kind == "macro"`
+
+A `macro_rules!` declarative macro. Contains a single string with the source representation of
+the macro with the patterns stripped, for example:
```

Comment:
What about macros 2.0?

Reply:
Forgot they were a thing. Due to my unfamiliarity with those and proc macros I've left them out for now.